### ISSUE TITLE
Setting Rotation in inspector is inconsistent with setting it using rotation_degrees in a script

### ISSUE THREAD

**Issue by tileuser** [godotengine/godot#76573]
### Godot version

4.0.2 stable mono

### System information

Windows 10

### Issue description

I'm am not sure why but setting the rotation as 90 degrees in the inspector gives a slightly different value compared to setting the rotation using `rotation_degrees` in a script. Doing `rotation = deg_to_rad(90)` and `rotation = (90 * PI) / 180` also gives the same values as doing `rotation_degrees = 90`.

![Screenshot showing inconsistency](https://user-images.githubusercontent.com/69766972/235289855-ce5b26f2-dc34-4ef6-ae6e-09ee914f0805.png)

This inconsistency also applies to other angle values. (45, 60, 180)

![Screenshot showing inconsistency - Figure 4](https://user-images.githubusercontent.com/69766972/235294625-d3573637-4c77-4551-b5f9-ec4853c9fa94.png) ![Screenshot showing inconsistency - Figure 2](https://user-images.githubusercontent.com/69766972/235294595-c12685f2-5702-4590-83a3-d2e1f836726b.png) ![Screenshot showing inconsistency - Figure 3](https://user-images.githubusercontent.com/69766972/235294610-f9eb6bfb-4d9b-48c2-934f-4b67ea6a8d5c.png)

I expected that that inputting 90 degrees in the inspector would yield the same result as setting `rotation_degrees = 90` in code. However, it does not.

### Steps to reproduce

- Create a new project
- Add a Node2D to the scene tree
- Select it and in the inspector, under the Node2D's Transform property, set its Rotation to 90
- Create and add a script to the Node2D that sets `rotation_degrees = 90` in the ready function
- Write `print(rotation)` and `print(rotation_degrees)` above and below the `rotation_degrees = 90` statement
- Run project to see and compare the printed out values of the node's Rotation between setting it in the inspector and setting it in a script

### Minimal reproduction project

[Inconsistency with setting Rotation in Inspector (4-29-2023).zip](https://github.com/godotengine/godot/files/11358717/Inconsistency.with.setting.Rotation.in.Inspector.4-29-2023.zip)

**Comment by Zireael07** [comment]
Never expect the values to be exact when you're using floating numbers. This is known and documented and a common pitfall, not just of Godot, of programming in general

**Comment by kleonc** [comment]
> Never expect the values to be exact when you're using floating numbers.

@Zireael07 I disagree, it's absolutely fine to expect exactly the same floating point calculations to produce exactly the same results.

---


> I'm am not sure why but setting the rotation as 90 degrees in the inspector gives a slightly different value compared to setting the rotation using `rotation_degrees` in a script. Doing `rotation = deg_to_rad(90)` and `rotation = (90 * PI) / 180` also gives the same values as doing `rotation_degrees = 90`.
> 
> ![Screenshot showing inconsistency](https://user-images.githubusercontent.com/69766972/235289855-ce5b26f2-dc34-4ef6-ae6e-09ee914f0805.png)

The reason of this behavior is that the scene is saved as a text file (`.tscn`) where floats are stored as strings only up to some decimal places of precision. Contents of the `node_2d.tscn` from the MRP:
```
[gd_scene load_steps=2 format=3 uid="uid://dwy6lwmrxc5ev"]

[ext_resource type="Script" path="res://ShowRotationInconsistency.gd" id="1_3yls2"]

[node name="Node2D" type="Node2D"]
rotation = 1.5708
script = ExtResource("1_3yls2")
```

So the original value of `deg_to_rad(90.0)` (approx `1.5707963705062866`) is saved with much less precision as a `"1.5708"` string. Then when loading such scene `"1.5708"` is parsed as a float which results in approx `1.5707999467849731`. Hence the discrepency with the original value.

Saving the scene in a binary format (`.scn`) makes the floats be saved as exact values, so no such issue occurs in such case.

I'm not sure whether saving floats as text was deliberately done with such small precision (to potentially decrease the file sizes?), or whether it was overlooked, or whether it was done for whatever other reason.

**Comment by Calinou** [comment]
> I'm not sure whether saving floats as text was deliberately done with such small precision (to potentially decrease the file sizes?), or whether it was overlooked, or whether it was done for whatever other reason.

We could save floats with full precision for scalar and vector floats, but use a lower precision for PackedFloat32Array and PackedFloat64Array.

**Comment by tileuser** [comment]
Ah, I see. The floats being stored as a string with less precision also seems to have been the cause of these previous issues as well: #20616, #29882, #74313, #75615.

**Comment by locatellidaniel** [comment]
I think the issue #59577 might be related too...

For PhysicalBone3D the Transform3D change after every save, suggesting a loss of precision:

![image](https://github.com/godotengine/godot/assets/1318427/8aab6d34-f15e-49a0-9883-1b113f8b7144)

I'll try the workaround of saving these scenes as binary, that should solve the issue of the Bones changing positions after every save, but of course binary files will not help to identify changes on source control.
